### PR TITLE
Use `wrap_patch` in autologging integrations to preserve original function attributes

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

## What changes are proposed in this pull request?

- A follow-up for #3294 
- Use `wrap_patch` in autologging integrations.

## How is this patch tested?

Verify no exisiting tests are broken.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [x] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/tensorflow.py [hunk 1]
```diff
@@ -1044,40 +1039,27 @@ def fit_generator(self, *args, **kwargs):
...
-        ),
-        gorilla.Patch(FileWriter, "add_summary", add_summary, settings=settings),
+        (EventFileWriter, "add_event", add_event),
+        (EventFileWriterV2, "add_event", add_event),
+        (tensorflow.estimator.Estimator, "train", train),
+        (tensorflow.keras.Model, "fit", fit),
+        (tensorflow.keras.Model, "fit_generator", fit_generator),
+        (tensorflow.estimator.Estimator, "export_saved_model", export_saved_model),
+        (tensorflow.estimator.Estimator, "export_savedmodel", export_savedmodel),
+        (FileWriter, "add_summary", add_summary),
```
Comment (harupy) [inline mlflow/tensorflow.py:1061]: @dbczumar Should we make the `name` argument of `wrap_patch` optional so we can omit it when a patch function has the same name as a target attribute?

```python
def wrap_patch(dest, patch, name=None, ...):
    name = patch.__name__ if name is None else name
    ...
```

Comment (dbczumar) [reply mlflow/tensorflow.py:1061]: I don't think that optimization is strictly necessary. `gorilla` doesn't do it, for example.

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @harupy !

File: mlflow/tensorflow.py (hunk 1 above)
Comment (harupy) [reply mlflow/tensorflow.py:1061]: makes sense

**Comment by harupy** [comment]
@dbczumar Thanks for the reivew! Merged!

**Pull request merged**